pkg/query/storeset.go

```diff
@@ -197,20 +231,64 @@ func (s *StoreSet) Update(ctx context.Context) {
 	s.mtx.Lock()
 	defer s.mtx.Unlock()
 
+	//Also record the number of occurrences of external label combinations.
+	externalLabelStores := map[string][]*storeRef{}
```

Comment:
can we have just `map[string]int` here (and in `StoreNode`'s `externalLabelStores` field)? Cannot see any `storeRef` pointer used anywhere